samples/cellular/lwm2m_carrier/src/main.c

```diff
@@ -12,6 +12,8 @@
 #include <lwm2m_carrier.h>
 #include "carrier_certs.h"
 
+static bool m_first_init;
```

Comment:
Why move this here? The goal of this variable is to avoid provisioning the certificates and the PSK on every modem_init (e.g. after a modem FOTA), which is unnecessary.